ui/tests/integration/components/ldap/page/roles-test.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupEngine } from 'ember-engines/test-support';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { render, click, fillIn } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { allowAllCapabilitiesStub } from 'vault/tests/helpers/stubs';
+
+module('Integration | Component | ldap | Page::Roles', function (hooks) {
+  setupRenderingTest(hooks);
+  setupEngine(hooks, 'ldap');
+  setupMirage(hooks);
+
+  hooks.beforeEach(function () {
+    this.server.post('/sys/capabilities-self', allowAllCapabilitiesStub());
+
+    this.store = this.owner.lookup('service:store');
+    this.store.pushPayload('secret-engine', {
+      modelName: 'secret-engine',
+      data: {
+        accessor: 'ldap_f3400dee',
+        path: 'ldap-test/',
+        type: 'ldap',
+      },
+    });
+    for (const type of ['static', 'dynamic']) {
+      this.store.pushPayload('ldap/role', {
+        modelName: 'ldap/role',
+        backend: 'ldap-test',
+        type,
+        ...this.server.create('ldap-role', type, { name: `${type}-test` }),
+      });
+    }
+    this.backend = this.store.peekRecord('secret-engine', 'ldap-test');
+    this.roles = this.store.peekAll('ldap/role');
+    this.breadcrumbs = [
+      { label: 'secrets', route: 'secrets', linkExternal: true },
+      { label: this.backend.id },
+    ];
+    this.promptConfig = false;
+
+    this.renderComponent = () => {
+      return render(
+        hbs`<Page::Roles @promptConfig={{this.promptConfig}} @backendModel={{this.backend}} @roles={{this.roles}} @breadcrumbs={{this.breadcrumbs}} />`,
+        { owner: this.engine }
+      );
+    };
+  });
+
+  test('it should render tab page header and config cta', async function (assert) {
+    this.promptConfig = true;
```

Comment:
I'm setting it to `true` before rendering the component so it just picks up the value and passes it into the component as an argument. This test validates that the config prompt is rendering as expected. It defaults to `false` in the `beforeEach` hook since none of the other tests need it.